rudder-core/src/main/scala/com/normation/rudder/services/queries/LdapQueryProcessor.scala

```diff
@@ -67,13 +68,17 @@ sealed trait ExtendedFilter
 final case class LDAPFilter(f:Filter) extends ExtendedFilter with HashcodeCaching
 
 //special ones
-sealed trait SpecialFilter extends ExtendedFilter
-final case class RegexFilter(attributeName:String, regex:String) extends SpecialFilter with HashcodeCaching
-final case class NotRegexFilter(attributeName:String, regex:String) extends SpecialFilter with HashcodeCaching
+sealed trait SpecialFilter  extends ExtendedFilter
+sealed trait RegexFilter    extends SpecialFilter
```

Comment:
Should the trait have attributeName and regex as def ? overriden by the two case classe to refactor some code ?

Reply:
we don't use that, and using it would mean even more refactoring I think. So I'm for letting things as they are, and wait for the addition of a JSON-query feature to see what need to be speficialized and how.